### PR TITLE
Improves sleeper code and makes syndie sleepers deconstructable

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This PR does four things:
1. Improves sleeper code by a lot, considering last time they were touched was years ago (last edit was 3 years ago when it was replaced with stasis units). This means a removal of ``scrambled_chems`` var in favor of just checking if the machine is emagged.
2. Removes the entirely unused ``sleep_console``.
3. Lowercases the dm file (its been bugging me forever, I didn't want to change it until I did a code improvement PR for it).
4. Lets Syndicate sleepers get deconstructed regardless of mapload.

## Why It's Good For The Game

For the Syndicate sleeper, I have 2 reasons for this:
1. Syndicate agents who accidentally destroy their sleepers can now rebuild it
2. People who successfully hijack the shuttle, as a nice bonus for people who manage to defeat nukies and take over their ship.

I personally see this as QoL but I can see it being labelled balance as well, if anyone wants to change that you can feel free.

## Changelog

:cl:
qol: Syndicate sleepers (the ones in the nukie base outpost) now drops their board on deconstruction.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
> Lowercases the dm file (its been bugging me forever, I didn't want to change it until I did a code improvement PR for it).

5 dollars says this causes the same case sensitive filename windows crap we deal with every time this happens

**Comment by Mothblocks** [comment]
oh wow look at that it already is. in your pr

**Comment by Mothblocks** [comment]
i think you should just not do the rename at all unless you pick a different filename like `sleepers.dm`

**Comment by JohnFulpWillard** [comment]
outsmarted the CI

**Comment by Mothblocks** [comment]
it's not the ci dude, it's *us* when we switch branches and git throws a fit on windows

**Comment by newfren** [comment]
there are syndicate sleepers in space in ruins with no combat in case that balance consideration is important. You can steal the regular sleeper board from charlie so it's probably whatever

**Review by Mothblocks — CHANGES_REQUESTED** [review]
rename it to `sleepers.dm` or something. this has absolutely fucked me over on my windows machine several times before and it drives me insane

File: code/game/machinery/sleepers.dm [hunk 1]
```diff
@@ -286,24 +321,41 @@
 	icon_state = "partypod"
 	base_icon_state = "partypod"
 	circuit = /obj/item/circuitboard/machine/sleeper/party
-	var/leddit = FALSE //Get it like reddit and lead alright fine
```
Comment (san7890) [inline code/game/machinery/sleepers.dm:289]: 😢 i'll miss u leddit

**Comment by Sealed101** [comment]
should probably mention the sleeper swaps on the ruin in cl

**Comment by JohnFulpWillard** [comment]
> should probably mention the sleeper swaps on the ruin in cl

Their functionality hasn't changed, they just look like NT ones instead of syndicate ones, if it's necessary I can add it but I feel like space ruins don't call for a changelog entry talking about how one item swapped colors

**Comment by newfren** [comment]
syndicate sleepers have different functionality to nt ones in that syndicate ones can be used by the person inside while nt ones cannot

**Comment by JohnFulpWillard** [comment]
> syndicate sleepers have different functionality to nt ones in that syndicate ones can be used by the person inside while nt ones cannot

I replaced them with ``/self_control`` subtype which does allow control from the inside

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/game/machinery/sleepers.dm [hunk 2]
```diff
@@ -161,21 +175,27 @@
 
 /obj/machinery/sleeper/process()
 	..()
-	check_nap_violations()
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/game/machinery/sleepers.dm:164]: It checks it in ``ui_act``, i thought it was weird to have it in process too, I can re-add it if you want me to.

File: code/game/machinery/sleepers.dm [hunk 3]
```diff
@@ -161,21 +175,27 @@
...
-	use_power(active_power_usage)
+	use_power(idle_power_usage)
 
 /obj/machinery/sleeper/nap_violation(mob/violator)
+	. = ..()
```
[the remark this answers is not in the document]
Comment (JohnFulpWillard) [reply code/game/machinery/sleepers.dm:181]: No, but it was added in case functionality is ever added (i hope not)

**Comment by Addust** [comment]
you should be able to get sleeper board from ruin sleepers but make the pried out boards have half max chemicals if you really think removing sleepers was a good idea

**Comment by JohnFulpWillard** [comment]
> you should be able to get sleeper board from ruin sleepers but make the pried out boards have half max chemicals if you really think removing sleepers was a good idea

sleepers don't have max chemicals

**Comment by Addust** [comment]
i mean the OD threshold limit, where if you spamclick a button it'll be greyed out, half the amount that the sleeper locks out the button on

**Comment by JohnFulpWillard** [comment]
That is still infinite omizine, and every other chem in there. I like them as a small reward for taking over syndicate areas, getting it from anywhere in space is really lame.

File: code/game/machinery/sleepers.dm [hunk 4]
```diff
@@ -15,25 +9,52 @@
...
 	var/efficiency = 1
+	///The minimum damage required to use any chem other than Epinephrine.
 	var/min_health = -25
-	var/list/available_chems
+	///Whether the machine can be operated by the person inside of it.
 	var/controls_inside = FALSE
+	///Whether this sleeper can be deconstructed and drop the board, if its on mapload.
+	var/deconstructable = FALSE
+	///Message sent when a user enters the machine.
+	var/enter_message = "<span class='notice'><b>You feel cool air surround you. You go numb as your senses turn inward.</b></span>"
```
Comment (SuperNovaa41) [inline code/game/machinery/sleepers.dm:24]:
```suggestion
	var/enter_message = span_boldnotice("You feel cool air surround you. You go numb as your senses turn inward.")
```

File: code/game/machinery/sleepers.dm [hunk 5]
```diff
@@ -98,35 +118,33 @@
...
 		return
-
 	close_machine(target)
 
-
 /obj/machinery/sleeper/screwdriver_act(mob/living/user, obj/item/I)
-	. = TRUE
-	if(..())
-		return
+	. = ..()
```
Comment (SuperNovaa41) [inline code/game/machinery/sleepers.dm:124]: you removed the `if (.)` check why

File: code/game/machinery/sleepers.dm [hunk 6]
```diff
@@ -286,24 +325,41 @@
 	icon_state = "partypod"
 	base_icon_state = "partypod"
 	circuit = /obj/item/circuitboard/machine/sleeper/party
-	var/leddit = FALSE //Get it like reddit and lead alright fine
-
 	controls_inside = TRUE
+	deconstructable = TRUE
+	enter_message = "<span class='notice'><b>You're surrounded by some funky music inside the chamber. You zone out as you feel waves of krunk vibe within you.</b></span>"
```
Comment (SuperNovaa41) [inline code/game/machinery/sleepers.dm:330]:
```suggestion
	enter_message = span_boldnotice("You're surrounded by some funky music inside the chamber. You zone out as you feel waves of krunk vibe within you.")
```

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]
looks fine other than these small things

File: code/game/machinery/sleepers.dm (hunk 5 above)
Comment (JohnFulpWillard) [reply code/game/machinery/sleepers.dm:124]: Parent is just a return, there was no point in having it.

**Pull request merged**